aws-analytics-pinpoint/src/test/java/com/amplifyframework/analytics/pinpoint/SessionClientTest.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.analytics.pinpoint
+
+import androidx.test.core.app.ApplicationProvider
+import com.amplifyframework.analytics.pinpoint.internal.core.idresolver.SharedPrefsUniqueIdService
+import com.amplifyframework.analytics.pinpoint.targeting.TargetingClient
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verifyOrder
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class SessionClientTest {
+
+    private val analyticsClientMock = mockk<AnalyticsClient>(relaxed = true)
+    private val targetingClientMock = mockk<TargetingClient>(relaxed = true)
+    private val uniqueIdService = mockk<SharedPrefsUniqueIdService>(relaxed = true)
+    private lateinit var sessionClient: SessionClient
+
+    @Before
+    fun setup() {
+        sessionClient = SessionClient(
+            ApplicationProvider.getApplicationContext(),
+            analyticsClientMock,
+            targetingClientMock,
+            uniqueIdService
+        )
+        every { uniqueIdService.getUniqueId() }.answers { "UNIQUE_ID" }
+    }
+
+    @Test
+    fun `test session start is recorded`() {
+        sessionClient.startSession()
+        Assert.assertNotNull(sessionClient.session)
+        Assert.assertTrue(sessionClient.session?.startTime!! < System.currentTimeMillis())
+        Assert.assertTrue(!sessionClient.session!!.isPaused())
+        verifyOrder {
+            analyticsClientMock.createEvent("_session.start", any(), any(), any(), any(), any(), any(), any(), any())
+            analyticsClientMock.recordEvent(any())
```

Comment:
Would like to see as many of these any's replaced with expected values if possible. Instead of any can you check for that the recorded event matches expectation. If you can't construct exact event, maybe use slot and check.

Reply:
Idea her is to just to test that on every start & stop session call corresponding eventTypes are recorded. Other record event tests are available in AnalyticsClientTest